cmd/agent/common/misconfig/global.go

```diff
@@ -8,19 +8,25 @@ package misconfig
 import "github.com/DataDog/datadog-agent/pkg/util/log"
 
 // ToLog outputs warnings about common misconfigurations in the logs
-func ToLog() {
-	for name, check := range checks {
-		if err := check(); err != nil {
-			log.Warnf("misconfig: %s: %v", name, err)
+func ToLog(agent string) {
+	for _, check := range checks {
+		if check.Agent == agent {
+			if err := check.Run(); err != nil {
+				log.Warnf("misconfig: %s: %v", agent, err)
+			}
 		}
 	}
 }
 
 type checkFn func() error
+type check struct {
+	Agent string
+	Run   checkFn
+}
 
-var checks = map[string]checkFn{}
+var checks = map[string]check{}
 
 // nolint: deadcode, unused
 func registerCheck(name string, c checkFn) {
-	checks[name] = c
+	checks[name] = check{Agent: name, Run: c}
```

Comment:
Here, `name` is the name of the check, not the agent it supports. You may need to add a third `supportedAgents` parameter.

Reply:
How do i know what to pass into `registerCheck` for `supportedAgents` in the init function?

Reply:
At the moment, it would be any agent that calls `ToLog`. I think that is just the process agent and the core agent, but I would double check to be safe